solutions/src/main/java/ru/sberbank/school/task08/JAXBManager.java

```diff
@@ -0,0 +1,78 @@
+package ru.sberbank.school.task08;
+
+import lombok.NonNull;
+import ru.sberbank.school.task08.state.*;
+import ru.sberbank.school.util.Solution;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import java.io.*;
+import java.util.List;
+import java.util.Objects;
+
+@Solution(8)
+public class JAXBManager extends SaveGameManager<JAXBMapState<GameObject>, GameObject> {
+
+    /**
+     * Класс-наследник должен иметь конструктор эквивалентный этому.
+     *
+     * @param filesDirectoryPath Путь до директории, в которой хранятся файлы сохранений
+     */
+    public JAXBManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+
+    }
+
+    @Override
+    public void saveGame(String filename, JAXBMapState<GameObject> gameState) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не должно быть null");
+        Objects.requireNonNull(gameState, "Сохраняемый объект не должен быть null");
+
+        try {
+            JAXBContext context = JAXBContext.newInstance(gameState.getClass());
+            Marshaller marshaller = context.createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+
+            marshaller.marshal(gameState, new File(filesDirectory + File.separator + filename));
+
+        } catch (JAXBException e) {
+            throw new SaveGameException("Возникла ошибка при записи объекта", e, SaveGameException.Type.IO, gameState);
+        }
+    }
+
+    @Override
+    public JAXBMapState<GameObject> loadGame(String filename) throws SaveGameException {
+        Objects.requireNonNull(filename, "Имя файла не должно быть null");
+        JAXBMapState savable = null;
+
+        try {
+            JAXBContext context = JAXBContext.newInstance(JAXBMapState.class);
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+
+            savable = (JAXBMapState) unmarshaller.unmarshal( new File(filesDirectory + File.separator + filename));
+        } catch (JAXBException e) {
+            throw new SaveGameException("Возникла ошибка при считывании объекта", e, SaveGameException.Type.IO, savable);
```

Comment:
Line is longer than 120 characters (found 121).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-208337903>*